### ISSUE TITLE
Structuring of list of classes using Julia for teaching

### ISSUE THREAD

**Issue by crstnbr** [JuliaLang/www_old.julialang.org#764]
The [list of classes using Julia](https://julialang.org/teaching/) has reached quite some length and should probably be structured in some way.

- AFAICT the list has no order. Probably we should just sort it lexicographically.
- It would be cool to have a (interactive) world map visualization of where Julia is being used for teaching.

**Comment by ViralBShah** [comment]
It's going to keep getting longer, so I'd rather just keep it as is for now. But a map seems like a good idea, and you can click to see. Also this list is based on submissions, and is probably quite incomplete. For example, on JuliaBox, we see logins from 600 universities (although not all may necessarily be using Julia for teaching classes).

**Comment by crstnbr** [comment]
I don't see how "it's going to keep getting longer" is a valid point against ordering. One just hast to check that new PR's don't break the order, right?

Regarding a map, I just hacked a quick and dirty version: https://github.com/crstnbr/juliamap

Suggestions and PR's are of course very welcome.

**Comment by crstnbr** [comment]
> Also this list is based on submissions, and is probably quite incomplete.

It's probably important to make it easier for people to add their institution to the list. I'd assume not everyone is familiar with PR's .

**Comment by ViralBShah** [comment]
You can generally just edit the file right here on github. Also, you are right. Let's try some new orderings.

**Comment by ViralBShah** [comment]
The map is pretty awesome. Can we embed it on the teaching/learning page? When you hover on a particular location, could it list the name of the university and the class?

**Comment by ChrisRackauckas** [comment]
Yeah, the map needs some information was taught at the school.

**Comment by crstnbr** [comment]
> You can generally just edit the file right here on github. Also, you are right. Let's try some new orderings.

Right, you can. But it says on the website: _"If you know of other classes using Julia for teaching, please submit a pull request to update this list."_ and it links to the PR page, not the file edit page. Maybe we should change the link.

But why not even add, "or write an email to teaching@julialang.org" (to be replaced by any meaningful email address) or similar to be safe? I'd assume there won't be many mails at a time.

Regarding the ordering, I'll make a PR that sorts the list lexographically soon.

**Comment by crstnbr** [comment]
> The map is pretty awesome. Can we embed it on the teaching/learning page?

Of course. I don't have docker though so I can't really clone the site and make and test the changes.

> When you hover on a particular location, could it list the name of the university and the class?

Makes sense. I'll update it asap.

**Comment by crstnbr** [comment]
Course information is now displayed as well.

**Comment by xorJane** [comment]
I love the map!

**Comment by crstnbr** [comment]
After #768 gets merged, the script https://github.com/crstnbr/juliamap/blob/master/juliamap.jl generates a `juliamap.html` file containing just the interactive map from the `teaching/index.md` file.

Maybe one should just embed this `juliamap.html` per iframe on the teaching/learning page. Every once in a while the script should be run again to update the map.

**Comment by ViralBShah** [comment]
#768 is merged. @crstnbr would you be up for putting the rest together?

**Comment by crstnbr** [comment]
I would, but as I'm running Windows I can't really use Docker to build and test everthing locally. Don't know when I will be able to do it on a linux machine, sorry.

**Comment by ViralBShah** [comment]
Closing this year. All new discussion should be in the new repo.
